ocaml/database/redo_log.ml

```diff
@@ -542,7 +544,10 @@ let maybe_retry f log =
 (* Functions relating to the lifecycle of the block device I/O process. *)
 
 (* Close any existing socket and kill the corresponding process. *)
-let shutdown log =
+
+let shutdown_m = Mutex.create ()
```

Comment:
The log should have the name of the resource being used exclusively, in this case `!(log.pid)`.

We might want to drop mutex inside log `dying_processes_mutex`, it's only used here

Reply:
Why do we need a thread to adjust a counter in the first place?

Reply:
I don't think this lock can be removed. The update is happening in a separate thread.